Prototope/Behaviors.swift

```diff
@@ -0,0 +1,218 @@
+//
+//  Collisions.swift
+//  Prototope
+//
+//  Created by Saniul Ahmed on 07/02/2015.
+//  Copyright (c) 2015 Khan Academy. All rights reserved.
+//
+
+import Foundation
+
+//MARK: Public Behavior Types
+
+/** Protocol for types describing the "configuration" of a behavior. */
+public protocol BehaviorType {}
+
+/** Describes a behavior that just calls a closure/block on every heartbeat,
+    passing the host layer to the closure. */
+public struct BlockBehavior: BehaviorType {
+    let handler: Layer -> Void
+    
+    public init(handler: Layer->Void) {
+        self.handler = handler
+    }
+}
+
+/** Value type describing the configuration of a collision behavior, specifying
+ the kind of the interaction (the collision conditions which should trigger the handler)
+ the layer with which the host layer is supposed to be colliding and a handler function */
+public struct CollisionBehavior: BehaviorType {
+    public enum Kind {
+        case Entering
+        case Leaving
+    }
+    
+    let kind: Kind
+    let otherLayer: Layer
+    let handler: () -> Void
+    
+    public init(on kind: Kind, _ otherLayer: Layer, handler: ()->Void) {
```

Comment:
Got some feedback from the designers. They'd actually prefer that `CollisionBehavior` calls `handler` on all `kind`s of events but passes `kind` in—they said they imagined they'd have to create both an entering and a leaving behavior most of the time. What do you think?